python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -222,54 +225,62 @@ def _update(self):
             nodes_allowed_to_terminate = self._get_nodes_allowed_to_terminate(
                 sorted_node_ids)
 
+        nodes_to_keep = []
+
+        def keep_node(node_id) -> None:
+            # Update per-type counts.
+            tags = self.provider.node_tags(node_id)
+            if TAG_RAY_USER_NODE_TYPE in tags:
+                node_type = tags[TAG_RAY_USER_NODE_TYPE]
+                node_type_counts[node_type] += 1
+            nodes_to_keep.append(node_id)
+
+        def terminate_node(node_id, kind, why=None) -> None:
```

Comment:
type annotation for arguments + function level description:
generally we follow something like this for the documentation where 1 line description. Then a new line with more description if necessary, then args, and then returns:
https://github.com/ray-project/ray/blob/e1cd8580a0b4aee1fc47671b0fdfe9ea1b1a3c02/python/ray/autoscaler/_private/resource_demand_scheduler.py#L356

Reply:
thanks!